src/CalculatorUI/Utils/VisualTree.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Media;
+
+// Light C++/CX port of Microsoft.Toolkit.Uwp.UI.Extensions.VisualTree from the Windows Community toolkit
+// Original version here:
+// https://raw.githubusercontent.com/windows-toolkit/WindowsCommunityToolkit/master/Microsoft.Toolkit.Uwp.UI/Extensions/Tree/VisualTree.cs
+
+namespace Calculator.Utils
+{
+    /// <summary>
+    /// Defines a collection of extensions methods for UI.
+    /// </summary>
+    sealed class VisualTree
+    {
+        /// <summary>
+        /// Find descendant <see cref="Windows.UI.Xaml.FrameworkElement ^"/> control using its name.
+        /// </summary>
+        /// <param name="element">Parent element.</param>
+        /// <param name="name">Name of the control to find</param>
+        /// <returns>Descendant control or null if not found.</returns>
+        internal static FrameworkElement FindDescendantByName(DependencyObject element, string name)
+        {
+            if (element == null || name == null || name.Length == 0)
+            {
+                return null;
+            }
+
+            var frameworkElement = (element as FrameworkElement);
+            if (frameworkElement != null && name.Equals(frameworkElement.Name))
+            {
+                return frameworkElement;
+            }
+
+            var childCount = VisualTreeHelper.GetChildrenCount(element);
+            for (int i = 0; i < childCount; i++)
+            {
+                var result = FindDescendantByName(VisualTreeHelper.GetChild(element, i), name);
+                if (result != null)
+                {
+                    return result;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Find first descendant control of a specified type.
+        /// </summary>
+        /// <param name="element">Parent element.</param>
+        /// <param name="type">Type of descendant.</param>
+        /// <returns>Descendant control or null if not found.</returns>
+        private static DependencyObject FindDescendant(DependencyObject element, Type typeName)
+        {
```

Comment:
Seem that the logic is pretty the same as FindDescendantByName, then should we also need the validation like line 27?

Reply:
Nit: Since the logic of finding by type name is similar to by name, we can encapsulate it as a method and pass the core caparison as a lambda parameter.

Reply:
Won't fix. Same as above